### PR TITLE
C#: Guard against `AssociatedSymbol` not being an `IEventSymbol`

### PR DESCRIPTION
**Pull request by hvitved**
Apply same logic as for property/indexer accessors to account for cases where the associated event cannot be determined. I have not been able to reproduce such cases locally, though we have seen reports of it happening.

### CODE REVIEW

File: csharp/extractor/Semmle.Extraction.CSharp/Entities/EventAccessor.cs [hunk 1]
```diff
@@ -3,45 +3,46 @@
...
+        /// if there is no associated symbol.
         /// </summary>
-        private IEventSymbol? EventSymbol => Symbol.AssociatedSymbol as IEventSymbol;
+        public static IEventSymbol? GetEventSymbol(IMethodSymbol symbol) =>
+            symbol.AssociatedSymbol as IEventSymbol;
 
         public override void Populate(TextWriter trapFile)
         {
             PopulateMethod(trapFile);
             ContainingType!.PopulateGenerics();
```
Comment (tamasvajk) [inline csharp/extractor/Semmle.Extraction.CSharp/Entities/EventAccessor.cs:26]: Shouldn't we have a `PopulateModifiers(trapFile);` call here?

File: csharp/extractor/Semmle.Extraction.CSharp/Entities/EventAccessor.cs [hunk 2]
```diff
@@ -53,14 +54,14 @@ public override void Populate(TextWriter trapFile)
             Overrides(trapFile);
         }
```
Comment (tamasvajk) [inline csharp/extractor/Semmle.Extraction.CSharp/Entities/EventAccessor.cs:55]: Properties have the following too. Shouldn't it be added?

```
            if (Symbol.FromSource() && Block is null)
            {
                trapFile.compiler_generated(this);
            }
```

**Review by tamasvajk — COMMENTED** [review]
I've added some minor questions.

File: csharp/extractor/Semmle.Extraction.CSharp/Entities/EventAccessor.cs (hunk 1 above)
Comment (hvitved) [reply csharp/extractor/Semmle.Extraction.CSharp/Entities/EventAccessor.cs:26]: Apparently not:
```
error CS1609: Modifiers cannot be placed on event accessor declarations
```

File: csharp/extractor/Semmle.Extraction.CSharp/Entities/EventAccessor.cs (hunk 2 above)
Comment (hvitved) [reply csharp/extractor/Semmle.Extraction.CSharp/Entities/EventAccessor.cs:60]: Makes sense.

**Review by tamasvajk — APPROVED** [review]

**Review by tamasvajk — APPROVED** [review]

**Pull request merged**